php/libraries/NDB_Page.class.inc

```diff
@@ -812,23 +812,15 @@ class NDB_Page implements RequestHandlerInterface
     }
 
     /**
-     * Returns false if user does not have access to page.
+     * Returns true if user has access to this page.
      *
-     * Usage: the hasAccess() function in a child class should contain
-     * something like:
+     * You do not need to overload hasAccess() if there are no access restrictions.
      *
-     *     // create user object
-     *     $user =& User::singleton();
+     * @param \User $user The user whose access is being checked
      *
-     *     return $user->hasPermission('superuser');
-     *
-     * You do not need to overload hasAccess() if there are no access restrictions
-     *
-     * @note   overloaded function
      * @return bool
-     * @access private
      */
-    function _hasAccess()
+    function _hasAccess(\User $user) : bool
```

Comment:
can this be removed like the others if it only returns true? (hopefully last comment! :p)

Reply:
No, this is the base of the class hierarchy for pages, the others were removed because they inherit this one.